src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -99,6 +150,124 @@ public void OnDiagnosticsUnhandledException(HttpContext httpContext, Exception e
             this.OnException(httpContext, exception);
         }
 
+        private RequestTelemetry InitializeRequestTelemetryOnHttpRequestInStart(HttpContext httpContext)
+        {
+            var requestTelemetry = new RequestTelemetry();
+
+            if (Activity.Current == null)
```

Comment:
store Activity.Current in local variable: it's AsyncLocal and accessing nultiple times it is not completely cheap